### PR TITLE
SpdxExpression: Support validating against deprecated license identifiers

### PR DESCRIPTION
**Pull request by sschuberth**
Signed-off-by: Sebastian Schuberth <sebastian.schuberth@here.com>

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1211)

### CODE REVIEW

File: spdx-utils/src/main/kotlin/SpdxExpression.kt [hunk 1]
```diff
@@ -45,14 +65,15 @@ sealed class SpdxExpression {
...
         @JvmStatic
-        fun parse(expression: String) = parse(expression, false)
+        fun parse(expression: String) = parse(expression, Strictness.ALLOW_ANY)
 
         /**
-         * Parse a string into an [SpdxExpression]. If [strict] is enabled, not only the syntax is checked but license
-         * expressions are also semantically checked to be valid SPDX identifiers. Throws an [SpdxException] if the
-         * string cannot be parsed.
+         * Parse a string into an [SpdxExpression]. [strictness] defines whether only the syntax is checked
+         * ([ALLOW_ANY][Strictness.ALLOW_ANY]), or semantics are also check but deprecated license identifiers are
```
Comment (mnonnenmacher) [inline spdx-utils/src/main/kotlin/SpdxExpression.kt]: s/check/checked

File: spdx-utils/src/main/kotlin/SpdxExpression.kt [hunk 2]
```diff
@@ -84,10 +105,12 @@ sealed class SpdxExpression {
     abstract fun spdxLicenses(): EnumSet<SpdxLicense>
 
     /**
-     * Semantically validate this expression to only contain SPDX identifiers. This includes SPDX licenses, exceptions
-     * and license references. Throws an [SpdxException] if validation fails.
+     * Validate this expression. [strictness] defines whether only the syntax is checked
+     * ([ALLOW_ANY][Strictness.ALLOW_ANY]), or semantics are also check but deprecated license identifiers are
```
Comment (mnonnenmacher) [inline spdx-utils/src/main/kotlin/SpdxExpression.kt]: s/check/checked

File: spdx-utils/src/main/kotlin/SpdxExpression.kt (hunk 1 above)
Comment (sschuberth) [reply spdx-utils/src/main/kotlin/SpdxExpression.kt]: Done.

File: spdx-utils/src/main/kotlin/SpdxExpression.kt (hunk 2 above)
Comment (sschuberth) [reply spdx-utils/src/main/kotlin/SpdxExpression.kt]: Done.

**Pull request merged**